serverless/src/tracing.ts

```diff
@@ -79,58 +82,69 @@ export function getTracingMode(config: Configuration): TracingMode {
 
 export function enableTracing(tracingMode: TracingMode, lambdas: LambdaFunction[], resources: Resources): void {
   if (tracingMode === TracingMode.XRAY || tracingMode === TracingMode.HYBRID) {
-    log.debug("Enabling Xray tracing...");
-    const xrayPolicies = {
-      Effect: ALLOW,
-      Action: [PUT_TRACE_SEGMENTS, PUT_TELEMETRY_RECORDS],
-      Resource: ["*"],
-    };
-    log.debug(`Xray policies: ${xrayPolicies}`);
+    enableXrayTracing(lambdas, resources);
+  }
+  if (tracingMode === TracingMode.HYBRID || tracingMode === TracingMode.DD_TRACE) {
+    enableDatadogTracing(lambdas, tracingMode);
+  }
+}
 
-    lambdas.forEach((lambda) => {
-      const role = findIamRole(resources, lambda);
+function enableXrayTracing(lambdas: LambdaFunction[], resources: Resources): void {
+  log.debug("Enabling Xray tracing...");
+  const xrayPolicies = {
+    Effect: ALLOW,
+    Action: [PUT_TRACE_SEGMENTS, PUT_TELEMETRY_RECORDS],
+    Resource: ["*"],
+  };
+  log.debug(`Xray policies: ${xrayPolicies}`);
 
-      if (role === undefined) {
-        throw new MissingIamRoleError(
-          `No AWS::IAM::Role resource was found for the function ${lambda.key} when adding xray tracing policies`,
-        );
-      }
+  lambdas.forEach((lambda) => {
+    lambda.properties.TracingConfig = { Mode: ACTIVE };
```

Comment:
Moving it to the beginning, so the mode can always be set even if the role is not found.